### DISCUSSION TITLE
Experience with Laplace approximation + negative binomial?

### DISCUSSION THREAD

**Discussion by gerverska** [JenniNiku/gllvm#162] [Q&A] [answered]
Hi everyone! I'm curious to know whether any folks here (users, authors, devs, maintainers, etc.) have experience using the Laplace approximation method in conjunction with the negative binomial family?

Bert's been helping me out with parameterizing a model (#158), and part of his troubleshooting tree involves investigating the result of setting `method = 'LA'`, which I am currently running. Here's what's being run right now:
```
model.gxext <- gllvm(Y,
                     X,
                     formula = ~ year * site + year * region + site * region,
                     family = 'negative.binomial',
                     num.lv = 0,
                     studyDesign = studyDesign,
                     row.eff = ~ (1 | block / tree),
                     offset = offset,
                     gradient.check = T,
                     method = 'LA',
                     control.start = list(n.init = 3))
```
This has been running for a few hours now. I guess I'm somewhat confused by this result, given that the speed test in [Niku et al. 2017](https://link.springer.com/article/10.1007/s13253-017-0304-7) seems to show that the LA is faster than the VA. Also, for this dataset, the VA had generally taken less than a minute to converge, so I'm not sure if this expected behavior for the LA + NB.

I have a sample size of 1473 with only 14 taxa, and there are 491 `tree` effects nested in 8 `block` effects (see above `row.eff`), in case that helps us compare our experiences--more info in the discussion #158 .

What have y'all seen?

**Comment by BertvanderVeen** [comment, marked as answer]
There was a bug in the package a while back that slowed the Laplace approximation down considerably. Try reinstalling the package from the github repository directly, that should solve it.

**Reply by gerverska** [reply]
Hmm...it still seems like the LA is still running pretty slowly. I was able to get something that converged when reltol = default, but the gradient check indicated that reltol needed a decrease. Bumped it down to 1e-15 (which had worked for the VA during my testing), and now it seems like it's back to taking awhile (~ 1 hr at this point). I did increase max.iter to 10000 (just in case iterations were exceeded, due to the decrease in reltol), so I'm sure this is contributing to this, but it never took this long for the VA whenever I set max.iter this high.

**Reply by gerverska** [reply]
Eh, I've got a separate model being fit with method = VA that's also taking awhile, so I'm probably just hitting bad seeds, unfortunately...

**Reply by BertvanderVeen** [reply]
It depends a bit on how the VA approximation is set up, and how the code is implemented, but usually VA is faster. Especially under the assumption of independent VA distributions for the random row effects, as is default in gllvm.

Later articles by Jenni also emphasise thay VA is usually faster, IIRC.
